File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/CoachStudentConnectionUpdateDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.ConnectionStatus;
import com.example.oriengo.model.enumeration.RequestInitiator;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CoachStudentConnectionUpdateDTO {
    private Long coachId;
    private Long studentId;
    private ConnectionStatus status;
    private LocalDateTime respondedAt;
    private RequestInitiator requestedBy;
}

File: OrienGO-Frontend/src/app/model/dto/CoachStudentConnectionUpdate.dto.ts
import { ConnectionStatus } from "../enum/ConnectionStatus.enum";
import { RequestInitiator } from "../enum/RequestInitiator.enum";

export interface CoachStudentConnectionUpdateDTO {
  coachId: number;
  studentId: number;
  status: ConnectionStatus;
  respondedAt?: string; // LocalDateTime → string (ISO)
  requestedBy: RequestInitiator;
}